test/integration/scheduler/predicates_test.go

```diff
@@ -920,3 +926,165 @@ func TestNodePIDPressure(t *testing.T) {
 
 	cleanupPods(cs, t, []*v1.Pod{testPod})
 }
+
+// TestEvenPodsSpreadPredicate verifies that EvenPodsSpread predicate functions well.
+func TestEvenPodsSpreadPredicate(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.EvenPodsSpread, true)()
+	// Apply feature gates to enable EvenPodsSpread
+	defer algorithmprovider.ApplyFeatureGates()()
+
+	context := initTest(t, "eps-predicate")
+	cs := context.clientSet
+	ns := context.ns.Name
+	defer cleanupTest(t, context)
+	// Add 4 nodes.
+	nodes, err := createNodes(cs, "node", nil, 4)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+	for i, node := range nodes {
+		// Apply labels "zone: zone-{0,1}" and "node: <node name>" to each node.
+		labels := map[string]string{
+			"zone": fmt.Sprintf("zone-%d", i/2),
+			"node": node.Name,
+		}
+		if err = testutils.AddLabelsToNode(cs, node.Name, labels); err != nil {
+			t.Fatalf("Cannot add labels to node: %v", err)
+		}
+		if err = waitForNodeLabels(cs, node.Name, labels); err != nil {
+			t.Fatalf("Adding labels to node failed: %v", err)
+		}
+	}
+
+	pause := imageutils.GetPauseImageName()
+	tests := []struct {
+		name string
+		pod  *v1.Pod
+		pods []*v1.Pod
+		fits bool
+		want []string // nodes expected to schedule onto
+	}{
+		// note: naming starts at index 0
+		{
+			name: "place pod on a 1/1/0/1 cluster with MaxSkew=1, node-2 is the only fit",
+			pod: st.MakePod().Namespace(ns).Name("p").Label("foo", "").Container(pause).
+				SpreadConstraint(1, "node", hardSpread, st.MakeLabelSelector().Exists("foo").Obj()).
+				Obj(),
+			pods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("p0").Node("node-0").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p1").Node("node-1").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p3").Node("node-3").Label("foo", "").Container(pause).Obj(),
+			},
+			fits: true,
+			want: []string{"node-2"},
+		},
+		{
+			name: "place pod on a 2/0/0/1 cluster with MaxSkew=2, node-{1,2,3} are good fits",
+			pod: st.MakePod().Namespace(ns).Name("p").Label("foo", "").Container(pause).
+				SpreadConstraint(2, "node", hardSpread, st.MakeLabelSelector().Exists("foo").Obj()).
+				Obj(),
+			pods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("p0a").Node("node-0").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p0b").Node("node-0").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p3").Node("node-3").Label("foo", "").Container(pause).Obj(),
+			},
+			fits: true,
+			want: []string{"node-1", "node-2", "node-3"},
+		},
+		{
+			name: "pod is required to be placed on zone0, so only node-1 fits",
```

Comment:
Both node 0 and 1 are labeled "zone 0", so why does only node-1 fit?

Reply:
Because node 0 has one pod already (L1001).

Reply:
Thanks for explaining it to me.